src/main/java/com/quickstart/HttpURLConnection.java

```diff
@@ -1,12 +1,54 @@
 package com.quickstart;
 
+import org.jsoup.Jsoup;
+
+import java.io.IOException;
+import java.util.Set;
+import org.jsoup.nodes.*;
+import org.jsoup.select.*;
+
+
 public class HttpURLConnection {
 
-    public static final String URL_TO_CONNECT = "https://jsainsburyplc.github.io/serverside-test/site/www.sainsburys.co.uk/" +
-            "webapp/wcs/stores/servlet/gb/groceries/berries-cherries-currants6039.html";
 
+    private String urlString;
 
-    public static void getConnection() {
+    public String getUrlString() {
+        return urlString;
+    }
 
+    public void setUrlString(String urlString) {
+        this.urlString = urlString;
     }
+
+    public Document getConnection() throws IOException {
```

Comment:
instead of setting url .. pass url as parameter to getConnection() method..

Reply:
It's best practice to not change state of the class.. here you are changing state of the class by setting url..

Reply:
when two threads set the url to class ..results of the get connection will be un expected  ..